### PR TITLE
#2208 - bnd-indexer-maven-plugin fails if inputDir folder contains a folder

### PR DESCRIPTION
**Pull request by rotty3000**

### CODE REVIEW

**Comment by rotty3000** [comment]
@bjhargrave can you restart this appveyor build please? There was a cloudbees issue last night which caused many builds to fail. I restarted the travis ones already.

**Comment by rotty3000** [comment]
nvm, I found a reason to update the PR.

File: maven/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/LocalIndexerMojo.java [hunk 1]
```diff
@@ -68,13 +79,45 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+		try {
+			toIndex = Files.find(inputDir.toPath(), Integer.MAX_VALUE, (p, a) -> {
+				String path = p.toString()
+					.replace('\\', '/'); // normalize windows paths
+
+				return includePatterns.stream()
+					.filter(i -> i.matcher(path)
+						.matches())
+					.filter(i -> !excludePatterns.stream()
+						.filter(e -> e.matcher(path)
```
Comment (bjhargrave) [inline maven/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/LocalIndexerMojo.java]: This is a much more concise stream.

```
return includePatterns.stream()
  .filter(i -> i.matcher(path).matches())
  // short circuit stream on any include pattern match
  .findAny() 
  // discard if it matches any exclude pattern
  .filter(i -> !excludePatterns.stream()
      .anyMatch(e -> e.matcher(path).matches()))
  .isPresent();
```

Comment (rotty3000) [reply maven/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/LocalIndexerMojo.java]: .. nice!

File: maven/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/LocalIndexerMojo.java [hunk 2]
```diff
@@ -68,13 +79,41 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
 		logger.debug("Indexing dependencies in folder: {}", inputDir.getAbsolutePath());
+		logger.debug("Including files: {}", includes);
+		logger.debug("Excluding files: {}", excludes);
 		logger.debug("Outputting index to: {}", outputFile.getAbsolutePath());
 		logger.debug("Producing additional gzip index: {}", includeGzip);
 		logger.debug("URI paths will be relative to: {}", baseFile);
 
-		Set<File> toIndex = new HashSet<>();
-		toIndex.addAll(asList(inputDir.listFiles()));
+		List<Pattern> includePatterns = includes.stream().map(s -> Glob.toPattern(s)).collect(Collectors.toList());
```
Comment (bjhargrave) [inline maven/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/LocalIndexerMojo.java]: We should change these to use `AntGlob` once #2342 is merged.

**Pull request merged**